### ISSUE TITLE
[Bug]: [Tag] [high-contrast] - Text color issue on truncated tags with "high-contrast" variant in g100 theme

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#18062]
### Package

@carbon/react

### Package version

1.70.0

### React version

18.3.1

### Description

If a tag is long and switches to the truncated version then the tag displays an incorrect text color deviating from the high-contrast color scheme applied
![Image](https://github.com/user-attachments/assets/6a785613-c7fc-447c-b4e1-fc9080c41790)

### Reproduction/example

https://stackblitz.com/edit/github-6fprgk-u6iecd?file=src%2FApp.jsx

### Steps to reproduce

use a tag with `high-contrast` with a long text
```
<Tag type="high-contrast">
        This is a very, very, very, long tag This is a very, very, very, long text     
 </Tag>
```

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by MichCest** [comment]
One thing worth mentioning - the issue exists in all of the other themes as well, it just isn't quite as obtrusive in white or g10 as it is in g90 or g100.

![Image](https://github.com/user-attachments/assets/5839dcc7-f136-4552-b7ec-478c42b4400a)
